### ISSUE TITLE
city in 2.13.0 cannot handle a certain address 132.248.1.0

### ISSUE THREAD

**Issue by miniway** [maxmind/GeoIP2-java#181]
city function in 2.13.0 cannot handle a certain address 132.248.1.0

```
Caused by: java.nio.charset.MalformedInputException: Input length = 1
	at java.nio.charset.CoderResult.throwException(CoderResult.java:281)
	at java.nio.charset.CharsetDecoder.decode(CharsetDecoder.java:816)
	at com.maxmind.db.Decoder.decodeString(Decoder.java:184)
	at com.maxmind.db.Decoder.decodeByType(Decoder.java:159)
	at com.maxmind.db.Decoder.decode(Decoder.java:146)
	at com.maxmind.db.Decoder.decodeMap(Decoder.java:278)
	at com.maxmind.db.Decoder.decodeByType(Decoder.java:153)
	at com.maxmind.db.Decoder.decode(Decoder.java:146)
	at com.maxmind.db.Decoder.decode(Decoder.java:88)
	at com.maxmind.db.Reader.resolveDataPointer(Reader.java:256)
	at com.maxmind.db.Reader.getRecord(Reader.java:176)
	at com.maxmind.geoip2.DatabaseReader.get(DatabaseReader.java:271)
	at com.maxmind.geoip2.DatabaseReader.getOrThrowException(DatabaseReader.java:240)
	at com.maxmind.geoip2.DatabaseReader.city(DatabaseReader.java:328)
```
It works in 2.9.0.

**Comment by oschwald** [comment]
It sounds like your database is corrupt. What database and what is the release date for it? Are you storing your database in a JAR file?

**Comment by miniway** [comment]
I used the same data file which is not included in jar both for 2.9 and 2.13.

If the database is corrupted, should it fail at the both versions?

**Comment by oschwald** [comment]
The code that the exception is coming from has not changed between 2.9 and 2.13. In order to diagnose what is going on, I would need the database information I asked for and preferably a complete code sample that demonstrates the issue.

**Comment by miniway** [comment]
I downloaded the data-file yesterday (Mar 1st, 2020) again. But the symptom was the same. Due to the data size of database and dependency files, I've uploaded a reproducible code to the Dropbox https://www.dropbox.com/s/lgihaywxik91v98/geoip.tgz?dl=0

`9.sh` and `13.sh` will execute the code per each version.

**Comment by oschwald** [comment]
Thanks. I am able to reproduce the issue with your test case. We'll take a look

Could you remove the file from your Dropbox share given that it has a database in it?

**Comment by oschwald** [comment]
Thanks for reporting. This has been fixed in `maxmind-db` 1.3.1. `geoip2` 2.13.1 should pull in that fix.
